libr/anal/esil.c

```diff
@@ -2495,19 +2420,14 @@ static int esil_mem_lsleq_n(RAnalEsil *esil, int bits) {
 	return ret;
 }
 
-static int esil_mem_lsleq1(RAnalEsil *esil) {
-	return esil_mem_lsleq_n (esil, 8);
-}
-static int esil_mem_lsleq2(RAnalEsil *esil) {
-	return esil_mem_lsleq_n (esil, 16);
-}
-static int esil_mem_lsleq4(RAnalEsil *esil) {
-	return esil_mem_lsleq_n (esil, 32);
-}
-static int esil_mem_lsleq8(RAnalEsil *esil) {
-	return esil_mem_lsleq_n (esil, 64);
+static int esil_mem_lsleq(RAnalEsil *esil, const char *word) {
+	int bytes;
+	if (sscanf (word, "<<=[%d]", &bytes) != 1) {
```

Comment:
definitely not efficient way to get a number, why not pass int instead of const char *word?